chore(analytics): Add debugging section to analytics page (#12856)
getsentry/sentry-docs: develop-docs/development-infrastructure/analytics.mdx

### Testing your Analytics Event

> Testing analytics events are important to making sure the data you are looking at is accurate. Any additional analytic event should be tested before merging to make sure that the events are firing correctly (with the correct values at the right times). A common issue we have seen without testing include events firing multiple times when they should only fire once.

Reviewer: ```suggestion
It's important to test analytic events to ensure the data you are looking at is accurate. Any additional analytic event should be tested before merging to make sure that the events are firing correctly (with the correct values at the right times). A common issue we see when testing gets overlooked is events firing multiple times when they should only fire once.
```

}
```

After deploying your changes, you can open the Dev Tools in your browser and in the "Network" tab, search for the `event/` request. This will show the events being sent to Reload and Amplitude.

## Debugging

> If your analytics aren't showing up after you added it, you can't find an event you expect to be there, or something else is wrong, there are a few things you can do to figure out what went wrong.

Reviewer: ```suggestion
If your analytics aren't showing up after you added it, you can't find an event you expect to be there, or something else goes wrong, there are a few troubleshooting steps you can try:
```

- Follow the steps [above](https://docs.sentry.io/development-infrastructure/analytics/#testing-your-analytics-event) to confirm that your analytics event is sending correctly, with the correct parameters.

> - Check Amplitude for blocked events. Some existing events have been blocked. In Amplitude, go to the "Data" section in the sidebar. From there, navigate to "Events" and search for your event name. It will show up with status "Blocked" if blocked, which means events won't show up. Some events were blocked in favor of automatic route or button analytics.

Reviewer: ```suggestion
- Check Amplitude for blocked events: In Amplitude, go to the "Data" section in the sidebar. From there, navigate to "Events" and search for your event name. It will show up with status "Blocked" if blocked, which means events won't show up. Some events may be blocked in favor of automatic route or button analytics.
```

- For route analytics, confirm that the analytic event isn't being blocked with `useDisableRouteAnalytics`. Some components already had an analytic event so the route analytics were disabled.

> - Check the types of the data you are sending. Arrays aren't recommended data types to send (hard to query and some unexpected behavior). Try to remove those if you are using them.

Reviewer: ```suggestion
- Check the types of the data you are sending. Arrays aren't recommended data types to send (they can be hard to query and produce some unexpected behavior). Try to remove those if you are using them.
```

> - Remember there will always be some discrepency. Ad-blockers for example can block events from being sent. This could be a cause of why some numbers aren't adding up.

Reviewer: ```suggestion
- Remember there will always be some discrepency. Ad-blockers, for example, can block events from being sent. This could be a cause of why some numbers aren't adding up.
```
